src/main/java/core/basesyntax/operations/BalanceOperation.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class BalanceOperation implements OperationHandler {
+    @Override
+    public void apply(Storage storage, FruitTransaction transaction) {
+        if (storage == null) {
+            throw new IllegalArgumentException("Storage cannot be null.");
+        }
+        if (transaction == null) {
+            throw new IllegalArgumentException("Transaction cannot be null.");
+        }
+        if (transaction.getQuantity() < 0) {
+            throw new IllegalArgumentException("Quantity must be non-negative.");
+        }
+
+        storage.updateFruit(transaction.getFruit(), transaction.getQuantity());

```

Comment:
The `updateFruit` method name does not clearly reflect the operation being performed. Since this is a balance operation, the method should set the quantity of fruit to a specific value rather than updating it. Consider renaming the method to better reflect its purpose, such as `setFruitBalance`.